File: android/sdk/src/main/kotlin/com/clevertap/android/nativedisplay/bridge/NativeDisplayBridgeListener.kt
package com.clevertap.android.nativedisplay.bridge

/**
 * Listener for Native Display units delivered through the bridge.
 *
 * Implement this to receive notifications when new display units are
 * available — either from CleverTap Core SDK (auto-wire mode) or from
 * manual JSON input.
 */
interface NativeDisplayBridgeListener {

    /**
     * Called when new Native Display units have been loaded and parsed.
     *
     * @param units The list of successfully parsed display units
     */
    fun onNativeDisplaysLoaded(units: List<NativeDisplayUnit>)
}

File: ios/Sources/CleverTapNativeDisplay/Bridge/NativeDisplayBridgeListener.swift
//
//  NativeDisplayBridgeListener.swift
//  CleverTapNativeDisplay
//

// MARK: - Bridge Listener Protocol
// Client interface for receiving native display unit updates from the bridge

import Foundation

/// Protocol for receiving native display units from the bridge.
/// Clients implement this to be notified when new display units are available,
/// whether from CleverTap Core SDK auto-wire or manual JSON input.
public protocol NativeDisplayBridgeListener: AnyObject {
    /// Called when native display units are loaded or updated.
    /// - Parameter units: The array of parsed native display units.
    func onNativeDisplaysLoaded(_ units: [NativeDisplayUnit])
}
